dm/worker/status.go

```diff
@@ -81,25 +82,30 @@ func (w *Worker) Status(stName string) []*pb.SubTaskStatus {
 				lockID = lockInfo.ID
 			}
 			cu := st.CurrUnit()
+
 			stStatus = pb.SubTaskStatus{
 				Name:                name,
 				Stage:               st.Stage(),
-				Unit:                cu.Type(),
 				Result:              st.Result(),
 				UnresolvedDDLLockID: lockID,
 			}
 
-			// oneof status
-			us := st.Status()
-			switch cu.Type() {
-			case pb.UnitType_Check:
-				stStatus.Status = &pb.SubTaskStatus_Check{Check: us.(*pb.CheckStatus)}
-			case pb.UnitType_Dump:
-				stStatus.Status = &pb.SubTaskStatus_Dump{Dump: us.(*pb.DumpStatus)}
-			case pb.UnitType_Load:
-				stStatus.Status = &pb.SubTaskStatus_Load{Load: us.(*pb.LoadStatus)}
-			case pb.UnitType_Sync:
-				stStatus.Status = &pb.SubTaskStatus_Sync{Sync: us.(*pb.SyncStatus)}
+			if cu != nil {
+				stStatus.Unit = cu.Type()
+				if st.Stage() == pb.Stage_Running {
```

Comment:
we don't store `status` now

Reply:
why? if paused, we can not query status anymore?

Reply:
now our pause equal stop, if subtask was stopped, no status will show in our framework

Reply:
then, if the task paused when errors occurred, can we query the status?

Reply:
after task's paused
* if dm-worker not restarted, it would show status
* if dm-worker restarted, because we don't run subtask(it was paused), it would not show status